### ISSUE TITLE
normalizer polling issue

### ISSUE THREAD

**Issue by RdeHevia** [monsoon-load-testing/monsoon#38]

**Comment by RdeHevia** [comment]
I don't think the current polling algo can deal with this scenario that I think is quite common:

We are testing a website more complex than a request bin. The time to load the page won't be milliseconds but seconds. Let's imagine the webpage is loaded in 3s on average.

Let's imagine that the step has 1 user, 1 step with 2s step delay. 

1) stepStartTime=0s, timeStampJsonSaved = 3s,
2) stepStartTime=5s, timeStampJsonSaved = 8s,
3) stepStartTime=10s, timeStampJsonSaved = 13s,
4) stepStartTime=15s, timeStampJsonSaved = 18s,
timewindows: 0-7.5, 7.5-15, 15-22.5...
Step 1 and 2 fall within timewindow 0-7.5. However the json of step2) is generated t=8s, which is outside of the timewindow. Thus, the normalizer will only process step 1) for the time window 0-7.5.

Similarly, normalizer will only process step3 for the time window 7.5-15.

In summary, many data points might get lost or not process correctly if the webpage takes more than milliseconds in loading the content.

Solution ideas: 
1) Process files of current timestamp when there is at least 1 file with the next timestamp

**Comment by alexfdrover** [comment]
Currently being rewritten by S.C & M.P.H

**Comment by RdeHevia** [comment]
Test 100821-5users-2steps-1min-1000sleep: Passed


100821-5users-2steps-1min-20000delay-0sleep: Not passed
![image](https://user-images.githubusercontent.com/55640085/140854996-98da04fe-a648-4204-8dd5-a8f0cd16028b.png)
 Behavior:
- User1-Go to bin: 3 data points expected. Normalizer outputs the first 2. Normalizer doesn't output the last one
- Normalizer.js never finalizes and keep running forever
Additional comment: I've noticed there was a typo on line 155. I fixed and rerun the test but it didn't change the behavior.
![image](https://user-images.githubusercontent.com/55640085/140854862-2a4d74e2-294c-4fd1-9e78-ad56a8b1a471.png)

**Comment by minhphanhvu** [comment]
Don't think with 20_000 delay the test will pass. The initialOffset needs to be larger than the delay. I suggest testing it with initialOffset doubling (40_000) the delay.

But I think it has something with the fact that the normalizer never fanalizes.
